packages/cdktf-cli/src/lib/models/terraform-cloud.ts

```diff
@@ -298,7 +298,7 @@ export class TerraformCloud implements Terraform {
     const url = `https://${this.hostname}/app/${this.organizationName}/workspaces/${this.workspaceName}/runs/${result.id}`;
     sendLog(`Created speculative Terraform Cloud run: ${url}`);
 
-    const pendingStates = ["pending", "plan_queued", "planning"];
+    const pendingStates = ["pending", "plan_queued", "planning", "cost_estimating", "policy_checking"];
```

Comment:
I'm wondering whether we'd need to add other states ([docs](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#run-states)) as pending here, too.
For example `pre_plan_running` also exists and might happen.

Related to this: Do we also need to add `cost_estimated`? I imagine that our current implementation will only rarely catch this one, as TFC might directly jump to the next state after it. So that might be a race condition waiting to happen.. 😅

Does anyone know whether we have a flowchart somewhere that shows the order in which these states happen? (E.g. `planned` might always be the final one when planning, and we could just invert the logic here and wait for that state to be reached?)

Reply:
I think it might be hard to get the 'final' state, since it depends on your TFC plan. Cost Estimation is not available for the cheaper tiers I think (though the [pricing page](https://www.hashicorp.com/products/terraform/pricing) doesn't elaborate).

Reply:
Though, re-reading your comment, I agree. Having `planned` be the state we care about might be good enough.

Reply:
- Adding `pre_plan_running` sounds like a good idea
- `cost_estimated` can be a final state in the sense that if there are no policy checks, an apply run can be created on TFC
- `planned` state only shows that planning has finished, after it can come the cost estimation and policy checks.

Here is an example order of events:
```
plan-queueable-at: | 2022-11-02T14:12:03+00:00,
queuing-at: | 2022-11-02T14:12:04+00:00,
plan-queued-at: | 2022-11-02T14:12:04+00:00,
planning-at: | 2022-11-02T14:12:10+00:00,
cost-estimating-at: | 2022-11-02T14:13:48+00:00
planned-at: | 2022-11-02T14:13:48+00:00,
cost-estimated-at: | 2022-11-02T14:13:57+00:00,
```

I don't know if you have access to TFE/TFC codebase, but maybe it's a good place to search for logic on when a run can be applied.

Reply:
Ok I think I got a good approach: We need to check the `data.attributes.actions.is-confirmable` flag from the [GET /runs/:run_id](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#get-run-details) API. Is this something you could take over @mutahhir / @ansgarm ?